components/director/internal/domain/tenant/repository.go

```diff
@@ -155,6 +159,51 @@ func (r *pgRepository) List(ctx context.Context) ([]*model.BusinessTenantMapping
 	return items, nil
 }
 
+func (r *pgRepository) buildSearchCondition(fields []string, searchTerm string) string {
+	var stmtBuilder strings.Builder
+
+	lastIdx := len(fields) - 1
+
+	for idx, field := range fields {
+		if idx != lastIdx {
+			stmt := fmt.Sprintf(`%s ILIKE '%%%s%%' OR `, field, searchTerm)
+			stmtBuilder.WriteString(stmt)
+			continue
+		}
+
+		stmt := fmt.Sprintf(`%s ILIKE '%%%s%%'`, field, searchTerm)
+		stmtBuilder.WriteString(stmt)
+	}
+	return stmtBuilder.String()
+}
+
+func (r *pgRepository) ListPageBySearchTerm(ctx context.Context, searchTerm string, pageSize int, cursor string) (*model.BusinessTenantMappingPage, error) {
+	var entityCollection tenant.EntityCollection
+
+	selectColumns := strings.Join(insertColumns, ", ")
+	query := fmt.Sprintf(`SELECT DISTINCT %s
+			FROM %s
+			WHERE %s = $1 AND (%s)`, selectColumns, tableName, statusColumn, r.buildSearchCondition(searchColumns, searchTerm))
+
+	page, totalCount, err := r.pageableQuerierGlobal.ListWithQueryGlobal(ctx, query, []interface{}{tenant.Active}, externalNameColumn, pageSize, cursor, &entityCollection)
+	if err != nil {
+		return nil, errors.Wrap(err, "while listing tenants from DB")
+	}
+
+	items := make([]*model.BusinessTenantMapping, 0, len(entityCollection))
+
+	for _, entity := range entityCollection {
```

Comment:
You can extract this code in a separate function. You can see multipleFromEntities functions in application and runtime repositories for reference.